### PR TITLE
Adding tensordot for tensorflow frontend, addresses the issue #3734

### PR DESCRIPTION
**Pull request by fspyridakos**
Implemented tensordot to tensorflow frontend API
Test passes but nore sure about test implementation

### CODE REVIEW

**Comment by fspyridakos** [comment]
It seems I can't reference the issue #3734 in the development so I'll mention it here

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
Hey, this looks great!
I've requested a few changes and left comments on them which needs to be addressed before we can merge the PR! :)
Thanks!

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py [hunk 1]
```diff
@@ -176,3 +177,45 @@ def test_tensorflow_slogdet(
...
+        with_out=False,
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
+        fw=fw,
+        frontend="tensorflow",
+        fn_tree="tensordot",
+        x=x1,
+        x2=x2,
+        axes=axis,
+    )
```
Comment (zaeemansari70) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py:221]: There should be a line at the end of file which seems to be missing.

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py [hunk 2]
```diff
@@ -176,3 +177,45 @@ def test_tensorflow_slogdet(
         fn_tree="linalg.slogdet",
         x=np.asarray(x, dtype=input_dtype),
     )
+
+
+# tensordot
+@given(
+    dtype_x1_x2_axis=_get_dtype_value1_value2_axis_for_tensordot(),
```
Comment (zaeemansari70) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py:184]: The `_get_dtype_value1_value2_axis_for_tensordot()` is giving an error as it has a missing required positional argument: `available_dtypes` which causes the tests to fail.

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py [hunk 3]
```diff
@@ -8,6 +8,7 @@
 import ivy_tests.test_ivy.helpers as helpers
 import ivy.functional.backends.tensorflow as ivy_tf
 from ivy_tests.test_ivy.helpers import handle_cmd_line_args
+from ivy_tests.test_ivy.test_functional.test_core.test_linalg import _get_dtype_value1_value2_axis_for_tensordot
```
Comment (zaeemansari70) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py:11]: The number of  `characters` on this line increases the maximum limit, which is 88, that causes a lint failure.
Adding some of the `characters`  to a new line would fix it. :)

**Comment by fspyridakos** [comment]
I updated the PR let me know if there's more feedback!

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py [hunk 4]
```diff
@@ -176,3 +178,48 @@ def test_tensorflow_slogdet(
...
     )
+
+
+# tensordot
+@handle_cmd_line_args
+@given(
+    dtype_x1_x2_axis=_get_dtype_value1_value2_axis_for_tensordot(
+        available_dtypes=helpers.get_dtypes("numeric"),
+    ),
+    as_variable=st.booleans(),
```
Comment (CheezBallzPi) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py:189]: `as_variable` and `native_array` are handled by `@handle_cmd_line_args` and can be removed from `@given`.

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py [hunk 5]
```diff
@@ -176,3 +178,48 @@ def test_tensorflow_slogdet(
...
+        input_dtypes=dtype,
+        as_variable_flags=as_variable,
+        with_out=False,
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
+        fw=fw,
+        frontend="tensorflow",
+        fn_tree="tensordot",
+        x=x1,
+        x2=x2,
```
Comment (CheezBallzPi) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py:223]: Is this supposed to be `x2` or `y`? The frontend function uses `x` and `y` as its arguments.

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py [hunk 6]
```diff
@@ -176,3 +178,48 @@ def test_tensorflow_slogdet(
...
+        axes=axis,
+    )
```
Comment (CheezBallzPi) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py:225]: `name` is an optional parameter but still needs to be tested for.

**Review by CheezBallzPi — CHANGES_REQUESTED** [review]
Reviewed.

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py (hunk 6 above)
Comment (fspyridakos) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py:225]: I am struggling to find other tests that include `name` to get some reference. Can you suggest how I test for it?

Comment (CheezBallzPi) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py:225]: It seems we aren't testing for `name`. You can leave it out and ignore this comment.

**Comment by fspyridakos** [comment]
Updated! Let me know if there's anything else I should change!

**Comment by DanielWit** [comment]
I'm curious why this is being reviewed while my pr for this function is waiting 9 days already 
https://github.com/unifyai/ivy/pull/3604

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
This PR has some conflicts, kindly resolve them 👍

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py [hunk 7]
```diff
@@ -239,3 +241,46 @@ def test_tensorflow_pinv(
...
+    helpers.test_frontend_function(
+        input_dtypes=dtype,
+        as_variable_flags=as_variable,
+        with_out=False,
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
+        fw=fw,
+        frontend="tensorflow",
+        fn_tree="tensordot",
+        x=x,
```
Comment (zaeemansari70) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py:283]: You have to pass this as a numpy array with the dtype which you are unpacking, kindly refer to this in the docs if you have any confusion. https://lets-unify.ai/ivy/deep_dive/16_ivy_frontends_tests.html

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]

**Pull request merged**